### PR TITLE
Jb44416

### PR DESCRIPTION
**Pull request by martyone**

### CODE REVIEW

**Comment by martyone** [comment]
The clang code model needs to know the additional qmake args pretty accurately.

File: src/plugins/mer/merbuildconfiguration.cpp [hunk 1]
```diff
@@ -89,9 +90,15 @@ MerBuildConfiguration::MerBuildConfiguration(Target *target, Core::Id id)
         if (!isReallyActive())
             return;
         QTC_ASSERT(project(), return);
-        if (!project()->files(Project::AllFiles).contains(document->filePath())
-                || !document->filePath().toString().contains(QRegExp("\\.spec$|\\.yaml$")))
+        if (!document->filePath().toString().contains(QRegExp("\\.spec$|\\.yaml$")))
             return;
+        if (!project()->files(Project::AllFiles).contains(document->filePath())) {
+            if (document->filePath().isChildOf(project()->rootProjectDirectory())) {
+                MessageManager::write(tr("Warning: RPM *.spec (or *.yaml) file not registered to the project."),
```
Comment (vigejolla) [inline src/plugins/mer/merbuildconfiguration.cpp]: I don't understand, in which situation do we end up in here?

Comment (martyone) [reply src/plugins/mer/merbuildconfiguration.cpp]: When the .spec/.yaml file is not referred by the .pro file.

Comment (vigejolla) [reply src/plugins/mer/merbuildconfiguration.cpp]: In that case, wouldn't we already return from either !isReallyActive() or project() checks?

Comment (martyone) [reply src/plugins/mer/merbuildconfiguration.cpp]: Nope. In this context `this` is `MerBuildConfiguration`, not `IDocument` - I guess that is what confused you.

File: src/plugins/mer/merbuildconfiguration.cpp [hunk 2]
```diff
@@ -56,6 +56,9 @@ namespace {
 
 const char MER_VARIABLES_CACHE_FILENAME[] = ".mb2/qmake_variables.cache";
 
+// Avoid actions on temporary changes and asking questions too hastily
+const int UPDATE_EXTRA_PARSER_ARGUMENTS_DELAY_MS = 3000;
```
Comment (vigejolla) [inline src/plugins/mer/merbuildconfiguration.cpp]: I think there are some cases (e.g. when opening a project) when we should not wait. It's rather annoying when the dialog interrupts you when you had already started e.g. opening another project or something.

Comment (martyone) [reply src/plugins/mer/merbuildconfiguration.cpp]: Unfortunately there are  too many scenarios to consider. Opening a project is actualy one of those that need some sort of care - be it the simple delay or something more complicated. When a project is being opened there are temporary states when it is unconfigured or configured using the default kit or so.

Ideally we should avoid the necessity for asking anything here, but I don't have a better idea how to reliably collect the additional args now. We should think about this more a propose some future work.

Comment (vigejolla) [reply src/plugins/mer/merbuildconfiguration.cpp]: I agree. I'll write it down on my own notes so that I can bring it up again in the future :)

**Pull request merged**